src/excmds.ts

```diff
@@ -4345,35 +4347,231 @@ const KILL_STACK: Element[] = []
 
 */
 //#content
-export async function hint(option?: string, selectors?: string, ...rest: string[]): Promise<any> {
-    if (!option) option = ""
+export async function hint(...args: string[]): Promise<any> {
+    // Argument parser state
+    enum State {
+        Initial,
+        ExpectF,
+        ExpectFR,
+        ExpectCallback,
+        ExpectExcmd,
+    }
+
+    // Open mode: how to act on the selected hintable element
+    enum OpenMode {
+        Default,
+        Tab,
+        BackgroundTab,
+        Window,
+        WindowPrivate,
+        Highlight,
+        Images,
+        ImagesTab,
+        Kill,
+        KillTridactyl,
+        Scroll,
+        SaveResource,
+        SaveImage,
+        SaveAsResource,
+        SaveAsImage,
+        // TODO: What does this actually do? The doc is unclear
+        Semicolon,
```

Comment:
`-;` calls `.focus()` on the selected element and explicitly tells Tridactyl that it is a child of the part of the page you want to scroll - https://github.com/tridactyl/tridactyl/blob/ced62d5fca6ea5aa986975cf5cf1765844498363/src/excmds.ts#L4603

It also has a little hack that lets you focus images.

The point of calling focus is mostly to let you then press the "right click button" on your keyboard (I think it's called context menu?) but on some sites you'll get some sort of hover pop-up.

Reply:
I don't get what the "tells Tridactyl that it is a child of the part of the page you want to scroll". What part of the page are we talking about? Why does the element needs to be made a child of that part?

I guess if the goal is to make keyboard right-click doable... It's some kind of "FocusRightclick" action? Maybe it should be noted in the docs that it's the intended use, I didn't know about it, but I want to use it now that I do!

Reply:
It's probably just easier to show you - e.g. https://doc.rust-lang.org/reference/types.html has two scrollable panes. `;;` lets you change which pane j,k etc. scroll. Just pick an element (any element) in each respective pane.

I don't think it _has_ to be a child of the element you want to scroll but it's usually easier to find a child than it is to find the exact element.

Reply:
Oh I see, by setting the focus to a particular (visible) element this indeeds switches the target for scroll events to its first parent with overflow enabled; aside from enabling keyboard right-clicking on it. `ScrollFocus` sounds nice enough (at least for the hint command internals!)